roles/custom/matrix-bot-matrix-registration-bot/tasks/validate_config.yml

```diff
@@ -6,5 +6,10 @@
       You need to define a required configuration setting (`{{ item }}`).
   when: "vars[item] == ''"
   with_items:
-    - "matrix_bot_matrix_registration_bot_bot_access_token"
     - "matrix_bot_matrix_registration_bot_api_token"
+
+- name: Fail if now bot secret is defined
+  ansible.builtin.fail:
+    msg: >-
+      You need to define the bot secret to login.
```

Comment:
"bot secret" is not very descriptive. Perhaps pointing people to `matrix_bot_matrix_registration_bot_bot_password` would be better?

I'm actually confused as to what needs to be defined:

- this `fail` task checks for both `matrix_bot_matrix_registration_bot_bot_password` and `matrix_bot_matrix_registration_bot_api_token`. Perhaps defining at least one of them is good enough indeed?

- `matrix_bot_matrix_registration_bot_api_token` is defined in terms of `matrix_bot_matrix_registration_bot_bot_access_token`, so there's one layer of indirection here, causing a bit of confusion.

- the docs (`docs/configuring-playbook-bot-matrix-registration-bot.md`) only mention `matrix_bot_matrix_registration_bot_bot_access_token`. They don't mention `matrix_bot_matrix_registration_bot_api_token` and `matrix_bot_matrix_registration_bot_bot_password`

-----

Is it that the bot now supports **either** an access token (old way) **or** a user id + password (new/additional way)? If a password is provided, the bot will get its own access token on startup? Does it do some caching to avoid creating new sessions on every startup? How well does it work?

------

If the user id + password flow is supported, then that's much more convenient for users to use - there's no longer the need to log in as the bot manually and copy the access token from the session.

Moreover, the playbook can automatically create the bot's account, so not only can we **eliminate the "manual login" step**, but we can **also potentially eliminate the "manual registration" step**.

As an example, see how it's done for Postmoogle. The documentation file (`docs/configuring-playbook-bot-postmoogle.md`) only asks for a password (`matrix_bot_postmoogle_password`). There already is a default user id (`matrix_bot_postmoogle_login`), so this variable is merely mentioned for the benefit of people who may wish to change it.

Then, `group_vars/matrix_servers` does the rest (automatic user creation): https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/ddbbd42718b15172cdf409f2c1050362d42c3151/group_vars/matrix_servers#L4009-L4034